models/purchase_settings.py

```diff
@@ -0,0 +1,16 @@
+from odoo import models, fields
+
+class PurchaseOrderSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    lifespan = fields.Integer(default=0)
+
+    def set_values(self):
+        super(PurchaseOrderSettings, self).set_values()
+        self.env['ir.config_parameter'].sudo().set_param('purchase.order.enhancement.settings.lifespan', self.lifespan)
+
+    def get_values(self):
+        res = super(PurchaseOrderSettings, self).get_values()
+        lifespan = self.env['ir.config_parameter'].get_param('purchase.order.enhancement.settings.lifespan')
+        res.update({'lifespan': lifespan})
+        return res
```

Comment:
You can do this way, it's ok, need to refactor anything now. 
But there is a short way to do the same function:
`lifespan = fields.Integer(default=0, config_parameter='purchase.order.enhancement.settings.lifespan')`